### PR TITLE
Add a failover flag for clusters

### PR DESCRIPTION
**Pull request by asuahsahua**

### CODE REVIEW

File: connection/hash_ring.go [hunk 1]
```diff
@@ -114,5 +117,15 @@ func (myHashRing *HashRing) GetConnectionPool(command protocol.Command) (connect
 
 	hash = myHashRing.BitMask & hash
 	connectionPool = myHashRing.ConnectionPools[hash]
+
+	for myHashRing.Failover && !connectionPool.IsConnected() {
```
Comment (mikemfrank) [inline connection/hash_ring.go]: Will this still infinite loop if all machines in the pool are down?

Comment (asuahsahua) [inline connection/hash_ring.go]: You're right, I'll fix this up

**Comment by mikemfrank** [comment]
Is there a reason this is going in to master rather than the failover branch in the forcedotcom repo?

**Comment by asuahsahua** [comment]
I don't think there is a reason to push to a separate branch in that repo, travis-ci still runs on this branch

**Comment by mikemfrank** [comment]
Looks good

**Pull request merged**